flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -397,7 +397,7 @@ private Configuration generateClusterConfiguration(Configuration configuration)
 		return resultConfiguration;
 	}
 
-	private CompletableFuture<ApplicationStatus> shutDownAsync(
+	public CompletableFuture<ApplicationStatus> shutDownAsync(
```

Comment:
This method should be moved away from the comment section declaring "internal methods" in this class. Or even better, instead of changing the visibility of methods in this class, couldn't you expose them only through `public static` helper methods in a class that is only in the test package like `ClusterEntryPointTestUtils.startCluster(ClusterEntryPoint ep)`?

Reply:
What about letting `ClusterEntrypoint` implement `AutoCloseableAsync` and make this public?

Reply:
Hmm, I would leave that call up to you. My feeling is when somebody receives a reference of this class that is not allowed to call `shutDownAsync`, I would better not make it public at all to prevent that clients call it accidentally. If that is not the case, yes why not use the interface that we have for the concept.